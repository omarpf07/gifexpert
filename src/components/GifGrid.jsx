import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifItem } from './GifItem';

export const GifGrid = ({ category }) => {
  const { gifs, loading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {loading && <h2>Loading...</h2>}
      <div className='card-grid'>
        {gifs.map((gif) => (
          <GifItem {...gif} key={gif.id} />
        ))}
      </div>
    </>
  );
};
