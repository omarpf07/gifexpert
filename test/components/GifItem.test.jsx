import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Tests on <GifItem/>', () => {
  const title = 'DBZ';
  const url = 'https://dbz.com/dbz.jpg';

  test('should match with snapshot', () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test('should show the image and the ALT attribute indicated.', () => {
    render(<GifItem title={title} url={url} />);
    const { src, alt } = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test('should show the title in the component', () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
