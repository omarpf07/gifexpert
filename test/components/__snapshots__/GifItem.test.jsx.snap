// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifItem/> should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="DBZ"
      src="https://dbz.com/dbz.jpg"
    />
    <p>
      DBZ
    </p>
  </div>
</div>
`;
